common/model/bundle.c

```diff
@@ -236,6 +257,76 @@ void bundle_reset_indexes(bundle_transactions_t *const bundle) {
   }
 }
 
+void bundle_set_messages(bundle_transactions_t *bundle, signature_fragments_t *messages) {
+  iota_transaction_t *tx = NULL;
+  size_t index = 0;
+  tryte_t **msg = NULL;
+  size_t msg_len = 0;
+  tryte_t trytes_buff[NUM_TRYTES_SIGNATURE];
+
+  BUNDLE_FOREACH(bundle, tx) {
+    memset(trytes_buff, '9', NUM_TRYTES_SIGNATURE);
+    memset(tx->data.signature_or_message, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_6561);
+    msg = signature_fragments_at(messages, index);
+    if (msg == NULL) {
+      break;
+    }
+    msg_len = strlen((char *)*msg);
+    memcpy(trytes_buff, *msg, msg_len);
+
+    // trytes to flex_trits
+    flex_trits_from_trytes(tx->data.signature_or_message, NUM_TRITS_SIGNATURE, trytes_buff, NUM_TRYTES_SIGNATURE,
+                           NUM_TRYTES_SIGNATURE);
+
+    memset(tx->attachment.branch, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_243);
+    memset(tx->attachment.trunk, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_243);
+    memset(tx->attachment.nonce, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_81);
+
+    transaction_set_attachment_timestamp(tx, 0);
+    transaction_set_attachment_timestamp_lower(tx, 0);
+    transaction_set_attachment_timestamp_upper(tx, 0);
+    tx->loaded_columns_mask.attachment = MASK_ATTACHMENT_ALL;
```

Comment:
The name of the function`bundle_set_messages` is misleading as it suggests that we are only seting messages. Why do we need to reset all attachment fields as well ?